Adds Canonical Kubernetes e2e test results bucket

chuckbutler opened the pull request:
We've just started running the e2e gambit on a regular basis with
published results to the google storage bucket. This should enable
gubernator to ingest our test results to the federated dashboard

During this dev cycle we encountered https://github.com/kubernetes/test-infra/issues/1003 

We punted on using the `upload-to-gs.sh` script in favor of a discovery in the `kubernetes/hack/lib/gubernator.sh`. At first glance this appears to be a next iteration of the `upload-to-gcs.sh` work, and was a bit more fleshed out. 

As our e2e run process diverges from what the expected integration is in the federated testing doc, if contributing those job source files would be a welcome contribution we can certainly clean those up and contribute with another PR. Let us know if thats of interest.  



<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/1161)
<!-- Reviewable:end -->

ixdy commented:
/lgtm

mikedanese commented:
Watcha gonna do about it robot?

ixdy commented:
/lgtm

mikedanese commented:
Does this just need to merge or does it require a config push?

fejta commented:
Woo! Nice PR.

Please add your results to testgrid: https://github.com/kubernetes/test-infra/blob/master/testgrid/config/config.yaml#L875

You'll need to add the gcs prefix to a testgroup, and then add the new testgroup to one or more dashboards (feel free to create your own dashboard as well as add it to relevant existing ones). The testgrid change does not require a push. 

What do you mean when you say "enable gubernator to ingest our test results into the federated dashboard?" cc @rmmh

rmmh commented:
Here's an example test result: https://k8s-gubernator.appspot.com/build/canonical-kubernetes-tests/logs/e2e-node/1122_17:10:48.000

I assume "dashboard" is referring to the [Kubernetes 24-Hour Test Report](http://storage.googleapis.com/kubernetes-test-history/static/index.html)

chuckbutler commented:
@rmmh correct. We were looking to add our results to both gubernator and the 24h results dashboard.

Our end-game here is to give you daily reports on the CDK deployments across clouds, with end game being including whats in master (daily build of master, and whats in -stable as our release process moves a little slower than yours, by hours sometimes a days) 

I'll take the items called out by fejta and give this another go. Thanks for the feedback

chuckbutler commented:
I've re-worked and re-submit the changes (in addition to rebasing to ensure this applies cleanly) 

Ready for re-evaluation at your leisure.

mbruzek commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.kubernetes.io/lgtm.png" height="20" width="61"/>

---







*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/test-infra/1161#-:-KXC6M1BPLDoIZthWqyP:bnfp4nl)*
<!-- Sent from Reviewable.io -->

ixdy commented on testgrid/config/config.yaml:
```diff
@@ -874,6 +874,9 @@ test_groups:
   gcs_prefix: kube_azure_log/kubernetes.azure.nightly
+# Canonical Distribution of Kubernetes (contact: @chuckbutler on github)
+- name: canonical-kubernetes-e2e-gce
+  gcs_prefix: canonical-kubernetes-tests/gce/logs/e2e-node
```
I'd expect this path and `buckets.yaml` to match, but they don't. should `buckets.yaml` be updated to `gs://canonical-kubernetes-tests/gce/logs`?

chuckbutler replied:
good shout. I updated the path to reflect what is in buckets.yaml.  I think the tree formula:

gss://bucket-name/logs/{cloud}/{test-profile}/

will work cleanly for this purpose. Thanks for calling this out.

ixdy commented on testgrid/config/config.yaml:
```diff
@@ -966,6 +969,11 @@ dashboards:
 
+- name: canonical-kubernetes
+  dashboard_tab:
+  - name: cdk-gce
```
what does the "d" stand for, out of curiosity?

chuckbutler commented:
@ixdy Good catch on the bucket pathing. I've given it some additional thought and documented the directory path and updated accordingly.

rmmh commented on buckets.yaml:
```diff
@@ -20,3 +20,6 @@ gs://kopeio-kubernetes-e2e/logs/:
   sequential: false
+gs://canonical-kubernetes-tests/logs/:
+  contact: "chuckbutler"
+  prefix: "juju:"
```
please add "sequential: false" here

rmmh commented on testgrid/config/config.yaml:
```diff
@@ -874,6 +874,9 @@ test_groups:
   gcs_prefix: kube_azure_log/kubernetes.azure.nightly
+# Canonical Distribution of Kubernetes (contact: @chuckbutler on github)
+- name: canonical-kubernetes-e2e-gce
+  gcs_prefix: canonical-kubernetes-tests/logs/gce/e2e-node
```
This layout is a little unusual-- can you put the logs in canonical-kubernetes-tests/logs/gce-e2e-node instead? If not, you should change the entry in buckets.yaml to be logs/gce as well, since it expects the layout to be like:

$PATH/$JOB/$BUILD/started.json

The 24-hour test history collector won't recursively search for job directories, so it won't handle your current results being stored in $PATH/gce/$JOB/$BUILD

chuckbutler commented:
@rmmh ok, Thanks for calling that out. I fixed our separator so it's now a flat structure under `$PATH`. That was a pretty obvious mixup on my part.

I think this is ready for re-review. I've adjusted our ci-job, and moved the requisite files in the bucket.

rmmh commented:
Thanks! They should show up on [24-hour test history page](http://storage.googleapis.com/kubernetes-test-history/static/index.html) within the hour.

marcoceppi commented:
@rmmh I just checked the page and I can't seem to find it, is there anything else missing for this?

rmmh commented:
As discussed on slack, they're changing the job names to be `kubernetes-e2e-gce` instead of `e2e-gce`. This will make them show up.
